lib/flow_builder.dart

```diff
@@ -92,21 +92,21 @@ class _FlowBuilderState<T> extends State<FlowBuilder<T>> {
   @override
   void didUpdateWidget(FlowBuilder<T> oldWidget) {
     super.didUpdateWidget(oldWidget);
+
+    final _shouldDispose = oldWidget.controller == null;
+
+    _removeListeners(dispose: _shouldDispose);
```

Comment:
Also, I don't think we should remove listeners when the state and controller haven't changed. Let me know what you think 👍

Reply:
Same here, sorry!